### DISCUSSION TITLE
Hubs Cloud cannot AWS deploy in Tokyo but ok in Oregon

### DISCUSSION THREAD

**Discussion by josedc** [mozilla/hubs#2995] [Hubs Cloud Questions] [answered]
I'm in Singapore and assume the closest Hubs Cloud supported region is Asia Pacific (Tokyo) ap-northeast-1.

I tried deploying twice on 22 August in Tokyo (my stacks named "kglam-1" and "kglam-2") and but failed both times with the following errors:

```
2020-08-22 17:39:22 UTC+0800	AppDb	CREATE_FAILED	Your subnet group only has one Availability Zone ap-northeast-1b that is supported by Aurora Serverless. Please ensure the subnet group includes at least two of the following Availability Zones, supported by Aurora Serverless: ap-northeast-1b, ap-northeast-1c, ap-northeast-1d. (Service: AmazonRDS; Status Code: 400; Error Code: InvalidVPCNetworkStateFault; Request ID: e0929912-df50-4f21-9702-aac3b3967112; Proxy: null)

2020-08-22 17:39:26 UTC+0800	kglam-2	ROLLBACK_IN_PROGRESS	The following resource(s) failed to create: [StackTopicHandlerRole, RouteTablePublicBInternetRoute, RouteTableAssociationAPrivate, SubnetNetworkAclAssociationBPublic, RouteTableAssociationBPrivate, InternalZoneSSLCert, AppDb, SpeelycaptorScratchBucket, AssetsBucket, BoxKeysBucket, SubnetNetworkAclAssociationAPublic, RouteTablePublicAInternetRoute, RootRedirectorBucket, RouteTableAssociationAPublic, SubnetNetworkAclAssociationBPrivate, RouteTableAssociationBPublic, SubnetNetworkAclAssociationAPrivate, NearsparkApiGatewayRestApi, LinkRedirectorBucket]. . Rollback requested by user.
```

I subsequently deployed the same config of HC successfully (my stack named "kglam-3") in US West (Oregon )us-west-2 with no errors or warning messages. I'm currently prototyping with that instance of HC.

Recenly @robin-k-wilson mentioned to document the AWS CloudFormation problems I experienced with the Tokyo region. This is also interesting because @doiskoh mentioned he is also in Singapore and was able to deploy HC in Tokyo with no CF errors. 

Just to double check, I just attempted to deploy HC again in Tokyo and got the same error as before. The below screenshot shows the error when I tried to create my stack "kglam-4" in Tokyo.

![Annotation 2020-09-10 102746](https://user-images.githubusercontent.com/171154/92676153-93756180-f353-11ea-8ee9-c03d2709bb04.jpg)

Any ideas on what I need to do to get it working in Tokyo would be appreciated. Thanks in advance!

**Comment by robinkwilson** [comment]
Thank you for the detailed write up! We're looking into this.

Quick questions before I write up the issue (thank you for your help!!):
1. Which domain recipe 1, 2, or 3 ([Domain Recipes](https://hubs.mozilla.com/docs/hubs-cloud-aws-domain-recipes.html)) are you following?
2. Do you have any custom AWS settings on this account that has to do with subnets? 
3. Are your domains/configuration fresh from Route 53?
4. Are you specifying something in the CloudFormation template to include subnets?

**Reply by emclaren** [reply]
Just a quick ping @josedc to see if you saw this message from Robin

**Comment by josedc** [comment]
Hi @robin-k-wilson and @emclaren ,

Thanks for following up. Sorry it's been a crazy week preparing our proposal using Mozilla Hubs for a community portal sponsored by the Singapore government. I was able to contact AWS Singapore for a Solution Architect to assist as I need to eventually get it running in Singapore.

Back to Robin's questions:

1. Which domain recipe 1, 2, or 3 (Domain Recipes) are you following?
I took the simplest approach using domain recipe 1 and registering completely new domains https://kglam.life and https://kglam.link on Route 53 for the Hubs instance.

2. Do you have any custom AWS settings on this account that has to do with subnets?
I don't think I did anything with subnets as I tried to follow the Hubs setup instructions step by step and had nothing previously in the Asia Pacific (Tokyo) ap-northeast-1 region nor the US West (Oregon) us-west-2 region. I am very new to AWS.

3. Are your domains/configuration fresh from Route 53?
Yes the domains did not exist prior to me creating them fresh on Route 53.

4. Are you specifying something in the CloudFormation template to include subnets?
I do not know how to read or change the CloudFormation and wouldn't know hoe to include subnets in the template. I didn't make any changes and tried to follow the tutorial step by step.

Hope the above helps.

Regards,
Jose

**Comment by robinkwilson** [comment, marked as answer]
Excellent! Well these tutorials deploy the stack for Recipe 1. 
Can you watch these video tutorials to double check that your domains are setup correctly?

[How to Deploy Hubs Cloud Part 1: Buying Domains](https://www.youtube.com/watch?v=2K7P8jFyHNc)
[How to Deploy Hubs Cloud Part 2: Deploying the Stack Recipe 1](https://www.youtube.com/watch?v=W0erzlDpo0U)

**Reply by robinkwilson** [reply]
Marking as answer for now to get this out of our pipeline. Please @ me if you need this opened.
